examples/objectdetection/src/App.tsx

```diff
@@ -37,35 +40,44 @@ const RenderTabBarIcon: React.FC<TabBarIconProps> = ({
 };
 
 function App() {
+  const [settings, setSettings] = React.useState<AppSettings>({
```

Comment:
The usual pattern for contexts is to have a `SettingsContextProvider` that wraps the state. That way you don't end up cluttering App.tsx with a bunch of state variables.

Reply:
Thanks, happy to do that, but having some difficulty figuring out what you're asking.  I am following this pattern: https://react.dev/reference/react/createContext#provider - I do have a SettingsContext object and I wrap the rest of the app with <SettingsContext.Provider />  Is there a way to wrap the state object and the SettingsContext into a a hook an reducee the code in App.tsx a bit? I am failing to find that pattern in the googles...